content/client-lib-development-guide/features.textile

```diff
@@ -1332,6 +1335,16 @@ h4. ChannelOptions
 ** @(TB2b)@ @cipher@, which is either:
 *** @(TB2b1)@ A @CipherParams@ instance, or
 *** @(TB2b2)@ an options hash (or language equivalent) consisting of any subset of @CipherParams@ fields that includes a @key@. In this case, the client library should call @getDefaultParams@, passing it the options hash, to obtain a @CipherParams@ instance
+*** @(TB2c)@ @params@ If provided these are sent to Ably in a new params property @(TR4n)@ of the ATTACH ProtocolMessage. Realtime will echo back the applied params in the ATTACHED if that field was specified in the ATTACH. Params could be:
+**** @(TB2c1)@ @modes@ comma-separated list of the channel subscription modes defined in @(TB2d)@ as a string.
+**** @(TB2c2)@ @delta@ If present this should have the value "vcdiff". Specifies the subscriber should receive data messages on the channel as deltas with the previous message.
+**** @(TB2c3)@ @rewind@ The rewind channel parameter relates to the initial attachment of a connection to a channel, and expresses the intent to attach to the channel at a position, or a point in time, in the past (that is, effectively “rewinding” the channel for the purposes of the present attachment). A rewind parameter can express a channel position in terms of a number of messages, or a time interval (ending in "s" or "m")
+**** @(TB2c4)@ @rewindLimit@ Sets a limit to the number of messages that will be returned if rewind is specified.
+*** @(TB2d)@ @modes@ array of channel subscription modes, e.g. "publish,subscribe,presence_subscribe". If present, these should overwrite the flags field @(TR3)@ of the ATTACH protocol message. 
+**** @(TB2d1)@ @publish@ Requests message publishing permission.
+**** @(TB2d1)@ @subscribe@ Requests message subscribing permission.
+**** @(TB2d1)@ @presence@ Requests presence message publishing permission.
+**** @(TB2d1)@ @presence_subscribe@ Requests presence message publishing permission.
```

Comment:
I don't understand why you've included this. The ChannelParams are already specified in https://github.com/ably/docs/pull/766/ , which you approved without comment -- if you disagreed with what was done in that PR then why would you not have commented there?

(FWIW most of this does not belong in the client library development spec: we went for a model where params are opaque to the library, so these spec items are mostly not things that are actually implementable by a client library developer. C.f. https://github.com/ably/docs/pull/766/files#diff-11900b395df266bc2bbfc1d11bdfc7a0R494 : `rewind` is only mentioned in the context of describing a method to test that the client library is successfully encoding the params)

Reply:
Yep, I approved this some time ago and did not really have it in mind when writing this one. Also true that `params` are opaque.